### PR TITLE
Add Doxygen documentation

### PR DESCRIPTION
**Pull request by tapaswenipathak**
Fixes https://github.com/STEllAR-GROUP/hpx/issues/1868.

## Proposed Changes

I added documentation for the listed functions. I think #1868 can be closed. `apply` and `apply_cb` are hpx/std functions, so I don't think Doxygen documentation should be added for them.

## Any background context you want to provide?
GSoD'19 applicant.

I am working on closing tickets labelled as documentation https://github.com/STEllAR-GROUP/hpx/labels/category%3A%20documentation.

### CODE REVIEW

File: hpx/runtime/runtime_mode.hpp [hunk 1]
```diff
@@ -33,8 +33,15 @@ namespace hpx
...
-    /// Get the internal representation (runtime_mode constant) from the
+		///////////////////////////////////////////////////////////////////////////
+		/// \brief Returns the internal representation (runtime_mode constant) from the
     /// readable string representing the name.
+		///
+		/// This represents the internal representation from the readable string
+		/// representing the name.
+		///
+		/// \param mode this represents the runtime mode
+
```
Comment (hkaiser) [inline hpx/runtime/runtime_mode.hpp]: Something went wrong with the indentation here.

Comment (hkaiser) [reply hpx/runtime/runtime_mode.hpp]: PS. please avoid using TABs, use spaces instead.

**Comment by tapaswenipathak** [comment]
@hkaiser do you think I can enhance anything else in this PR?

File: hpx/runtime/threads/thread_enums.hpp [hunk 2]
```diff
@@ -49,8 +49,13 @@ namespace hpx { namespace threads
                                  without high priority rescheduling */
     };
 
+    ///////////////////////////////////////////////////////////////////////////
+    /// \brief Returns the thread state
+    ///
```
Comment (msimberg) [inline hpx/runtime/threads/thread_enums.hpp]: This returns the *name* of the given state. Could you change the description to say that?

**Pull request merged**